controllers/zookeepercluster_controller.go

```diff
@@ -100,8 +100,11 @@ func (r *ZookeeperClusterReconciler) Reconcile(_ context.Context, request ctrl.R
 		}
 		return reconcile.Result{Requeue: true}, nil
 	}
+	if finalized, err := r.reconcileFinalizers(instance); err != nil || finalized {
```

Comment:
It would be much cleaner if we returned an error when there is a need to do another attempt to run a reconcile loop.